torch/csrc/DynamicTypes.cpp

```diff
@@ -79,8 +79,11 @@ THPLayout* getTHPLayout(at::Layout layout) {
   return thp_layout;
 }
 
-PyObject* createPyObject(const at::Storage& storage) {
-  if (storage.device_type() != at::DeviceType::Meta &&
+PyObject* createPyObject(
+    const at::Storage& storage,
+    bool always_create_python_storage) {
```

Comment:
This is a misnomer and I think it confused @albanD. You aren't creating a python subclass storage, you're just allowing for a (transient) Python object binding to the storage.

Reply:
But we only want to allow that for subclasses right?
Or are we saying that nullptr storage with non-zero size are rare enough that we're happy moving them from being an error to a weird looking storage object? If that is true, we can remove this flag altogether?

Reply:
Yes, I'm proposing we remove this flag altogether, and fixing the bindings to check for nullptr and avoid segfaulting.

Reply:
Confirming: you're saying that we're ok with unconditionally returning `Storage` object to the user in python even if its data pointer is null, but we just want any of the python-bound `Storage` API's to check for null and raise an error?

That sounds very reasonable

Reply:
I think you can also end up with a nullptr storage which also causes trouble, IIRC.